doc/reference/networking/net_l2.rst

```diff
@@ -20,38 +19,40 @@ object. The L2 layer dictates the API provided by the device driver,
 specific for that device, and optimized for working together.
 
 Currently, there are L2 layers for Ethernet, IEEE 802.15.4 Soft-MAC,
-Bluetooth IPSP, and a dummy one, which is a generic layer example that
-can be used as a template for writing a new one.
+Bluetooth IPSP, CANBUS, OpenThread, WiFi and a dummy one, which is a generic
+layer example that can be used as a template for writing a new one.
 
 L2 layer API
 ************
 
 In order to create an L2 layer, or even a driver for a specific L2
-layer, one needs to understand how the IP layer interacts with it and
-how the L2 layer is supposed to behave. The generic L2 API has 3
-functions:
-
-- recv: All device drivers, once they receive a packet which they put
-  into a :c:type:`struct net_pkt`, will push this buffer to the IP
-  core stack via :c:func:`net_recv_data()`. At this point, the IP core
+layer, one needs to understand how the L3 layer interacts with it and
+how the L2 layer is supposed to behave.
+See also :ref:`network stack architecture <network_stack_architecture>` for
```

Comment:
did you mean ``:ref:`ip_stack_architecture` `` ?

Reply:
I am anticipating the new version of #13714 which will introduce new ref name for that document. I will re-submit this PR again after that one is merged. I would like to get @tbursztyka comments to this one as he originally created this L2 document, he will be back from vacation next week.